soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java

```diff
@@ -0,0 +1,232 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dromara.soul.admin.listener.nacos;
+
+import com.alibaba.nacos.api.config.ConfigService;
+import com.alibaba.nacos.api.exception.NacosException;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.List;
+
+import org.dromara.soul.common.dto.AppAuthData;
+import org.dromara.soul.common.dto.MetaData;
+import org.dromara.soul.common.dto.PluginData;
+import org.dromara.soul.common.dto.RuleData;
+import org.dromara.soul.common.dto.SelectorData;
+import org.dromara.soul.common.enums.DataEventTypeEnum;
+import org.dromara.soul.common.utils.GsonUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * The TestCase for NacosDataChangedListener.
+ *
+ * @author cottom
+ * @date 2020/12/4
+ */
+@RunWith(MockitoJUnitRunner.class)
+public final class NacosDataChangedListenerTest {
+    private static final String GROUP = "DEFAULT_GROUP";
+
+    private static final String PLUGIN_DATA_ID = "soul.plugin.json";
+
+    private static final String SELECTOR_DATA_ID = "soul.selector.json";
+
+    private static final String RULE_DATA_ID = "soul.rule.json";
+
+    private static final String AUTH_DATA_ID = "soul.auth.json";
+
+    private static final String META_DATA_ID = "soul.meta.json";
+
+    private ConfigService configService;
+
+    private NacosDataChangedListener nacosDataChangedListener;
+
+    @Before
+    public void setUp() {
+        configService = new NacosMockConfigService();
+        nacosDataChangedListener = new NacosDataChangedListener(configService);
+    }
+
+    @Test
+    public void testOnAppAuthChanged() throws NacosException {
+        String mockAppKey = "MOCK_APP_KEY";
+        String mockAppKey2 = "MOCK_APP_KEY2";
+        String mockAppSecret = "MOCK_APP_SECRET";
+        AppAuthData appAuthData =
+                AppAuthData.builder().appKey(mockAppKey).appSecret(mockAppSecret).enabled(true).build();
+        AppAuthData appAuthData2 =
+                AppAuthData.builder().appKey(mockAppKey2).appSecret(mockAppSecret).enabled(true).build();
+        nacosDataChangedListener.onAppAuthChanged(
+                ImmutableList.of(appAuthData, appAuthData2), DataEventTypeEnum.CREATE);
+        Assert.assertEquals(
+                configService.getConfig(AUTH_DATA_ID, GROUP, 100),
+                GsonUtils.getInstance()
+                        .toJson(ImmutableMap.of(mockAppKey2, appAuthData2, mockAppKey, appAuthData)));
+        nacosDataChangedListener.onAppAuthChanged(
+                ImmutableList.of(appAuthData), DataEventTypeEnum.DELETE);
+        Assert.assertEquals(
+                configService.getConfig(AUTH_DATA_ID, GROUP, 100),
```

Comment:
Here it's easy to get confused. Why does the config still exist when we've deleted it. You should add more comment about what `NacosMockConfigService`  it really does.

Reply:
Why confused? We have `appAuthData ` and `appAuthData2 ` in the config. Now we have only delete the `appAuthData`, the left `appAuthData2` is as a matter of course @dengliming

Reply:
em my bad.